pulsar-broker-common/src/main/java/org/apache/pulsar/common/configuration/VipStatus.java

```diff
@@ -47,13 +47,33 @@ public class VipStatus {
     // log a full thread dump when a deadlock is detected in status check once every 10 minutes
     // to prevent excessive logging
     private static final long LOG_THREADDUMP_INTERVAL_WHEN_DEADLOCK_DETECTED = 600000L;
-    private static volatile long threadDumpLoggedTimestamp;
+    private static volatile long lastCheckStatusTimestamp;
+
+    // Since the status endpoint doesn't have authentication, it will be necessary to have a solution to prevent
+    // introducing a new DoS vulnerability where calling the status endpoint in a tight loop could introduce
+    // significant load to the system. One way would be to check that the deadlock check is executed only
+    // when there's more than 1 seconds from the previous check.
+    // If it's less than that, the previous result of the deadlock check would be reused.
+    private static final long DEADLOCK_DETECTED_INTERVAL = 1000L;
```

Comment:
Since the logic impacts the complete `checkStatus` result. I'd suggest renaming this field. It could also be considered to reduce the value from 1000 to 500.
```suggestion
    private static final long CHECK_STATUS_INTERVAL = 500L;
```

Reply:
> Since the logic impacts the complete `checkStatus` result. I'd suggest renaming this field. It could also be considered to reduce the value from 1000 to 500.

This is a great proposal, the meaning of the fields looks clearer and more specific and I will modify it.

Reply:
Completed